src/test/transaction_tests.cpp

```diff
@@ -174,9 +174,29 @@ unsigned int FillFlags(unsigned int flags)
 // Assumes that mapFlagNames contains all script verify flags.
 std::set<unsigned int> ExcludeIndividualFlags(unsigned int flags)
 {
+    std::vector<unsigned int> set_flags;
+    const size_t n_flags = sizeof(unsigned int);
+    set_flags.reserve(n_flags);
+    for (size_t i = 0; i < n_flags; ++i) {
+        const unsigned int flag = flags & (1 << i);
+        if (flag) {
+            set_flags.push_back(flag);
+        }
+    }
+    const size_t combos = 1 << set_flags.size();
+    // skip last all set flags
+    std::vector<unsigned int> all_flag_combos(combos-1);
+    for (size_t i = 0; i < combos-1; ++i){
+        for(size_t j = 0; j < set_flags.size(); ++j) {
+            if (i & (1<<j)) {
+                all_flag_combos[i] |= set_flags[j];
+            }
+        }
+    }
+
     std::set<unsigned int> flags_combos;
-    for (const auto& pair : mapFlagNames) {
-        const unsigned int flags_excluding_one = TrimFlags(flags & ~(pair.second));
+    for (const unsigned int excluded_flags : all_flag_combos) {
+        const unsigned int flags_excluding_one = TrimFlags(excluded_flags);
         if (flags != flags_excluding_one) {
             flags_combos.insert(flags_excluding_one);
         }
```

Comment:
Not sure if this is a problem, but notice that `TrimFlags` might cause the resulting `flag_combos` to have duplicate elements.

Eg, if `SCRIPT_VERIFY_P2SH` and `SCRIPT_VERIFY_WITNESS` are set in `flags`, the mapping done by `TrimFlags` will make 25% of the resulting elements duplicates.

Reply:
yeah I could do a de-dup pass with TrimFlags. It makes more sense IMO to keep TrimFlags even if superfluos with the above as TrimFlags is a separate validity of flag combo check v.s. the dumb subset generator.